app/views/devise/registrations/new.html.erb

```diff
@@ -13,6 +13,11 @@
       <%= f.label :username %><br />
       <%= f.text_field :username, autofocus: true, autocomplete: "username" %>
     </div>
+
+    <div class="field">
+      <%= f.label :avatar, "Profile photo" %><br />
+      <%= f.file_field :avatar, autofocus: true, autocomplete: "username" %>
+    </div>
```

Comment:
Maybe we can remove this avatar field here and have this avatar have a default image available upon successful signup instead. Then allow the users to update the image in their user's profile. Just a suggestion though.

Reply:
Image upload upon sign up is optional. User will get a randomized default avatar from a choice of 8 photos. The random avatar is made sure to be to the user so that he/she will get the same one after he/she logs in again. Also, users are free to update their avatar in the edit profile page.